src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -144,6 +144,12 @@ public override void OnStartActivity(Activity activity, object payload)
                 {
                     activity.SetTag(SemanticConventions.AttributeHttpUserAgent, userAgent);
                 }
+
+                var xForwardedFor = request.Headers["X-Forwarded-For"].FirstOrDefault();
+                if (!string.IsNullOrEmpty(xForwardedFor))
+                {
+                    activity.SetTag(SemanticConventions.AttributeHttpClientIP, xForwardedFor.Split(',').First().Trim());
```

Comment:
Do we want this on by default, or an opt-in (e.g. via some config while add the instrumentation)?
I guess a more general question is - what's the bar for the default vs opt-in tags.

Reply:
IMO anything that requires some manipultion (perf-impacting) shoud be opt-in.

Reply:
@reyang @CodeBlanch Those attributes are optional based on the spec, so it totally make senses to opt-in via instrumentation configuration. But this way customer may have to do some extra work when instrumenting. I believe it would be best if collecting these attributes can be done by SDK by default for all vendors to use similar to other languages. :relaxed:

Reply:
The concern is: 
Its an optional attribute per spec's semantic convention.
There is some cost involved (string alloc/extract etc)
People who don't care about this option attribute shouldn't be paying the cost for it.

If the setting is part of AspNetInstrumentationOptions, then opting in to it straightforward.

Reply:
My vote would be:
1. For things that are marked as "required" by the spec, we turn it on by default, and there is no way to opt-out unless the spec asks for such switch.
2. For things that are marked as "optional" by the spec, we use our best judgement based on the cost/return, most of the fields should be off by default, and allow people to opt-in via configuration.
3. For things that are not in the spec, if we see the need, add it to the spec and come back to the implementation.

Reply:
@reyang @cijothomas @CodeBlanch Hey guys, thanks for the suggestions! Do you recommend adding different option for specific attribute? Like adding `EnableHttpClientIP`(or other option name) specifically for recording `http.client_ip`, or adding one general option for recording all attributes that are not default?

Reply:
Just go with `RecordHttpServerAttributes` as `http.client_ip` belongs to [http server attributes](https://github.com/open-telemetry/opentelemetry-specification/blob/a2758014f408f64ff84728918d671ee3fdab2225/specification/trace/semantic_conventions/http.md#http-server-semantic-conventions).